File: src/test/java/org/oewntk/xml/in/TestsXmlModelDuplicates.java
/*
 * Copyright (c) 2021. Bernard Bou.
 */

package org.oewntk.xml.in;

import org.junit.BeforeClass;
import org.junit.Test;
import org.oewntk.model.LibTestModelDuplicates;
import org.xml.sax.SAXException;

import javax.xml.parsers.ParserConfigurationException;
import java.io.IOException;

public class TestsXmlModelDuplicates
{
	@BeforeClass
	public static void init()
	{
		TestsXmlCommon.init();
	}

	@Test
	public void testKeyOEWN()
	{
		LibTestModelDuplicates.testDuplicatesForKeyOEWN(TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test(expected = AssertionError.class)
	public void testKeyPos()
	{
		LibTestModelDuplicates.testDuplicatesForKeyPos(TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testKeyIC()
	{
		LibTestModelDuplicates.testDuplicatesForKeyIC(TestsXmlCommon.model, TestsXmlCommon.ps);
	}

	@Test
	public void testKeyPWN()
	{
		LibTestModelDuplicates.testDuplicatesForKeyPWN(TestsXmlCommon.model, TestsXmlCommon.ps);
	}
}

File: src/test/java/org/oewntk/xml/in/TestsXmlModelDuplicates.kt
/*
 * Copyright (c) 2021. Bernard Bou.
 */
package org.oewntk.xml.`in`

import org.junit.BeforeClass
import org.junit.Test
import org.oewntk.model.LibTestModelDuplicates.testDuplicatesForKeyIC
import org.oewntk.model.LibTestModelDuplicates.testDuplicatesForKeyOEWN
import org.oewntk.model.LibTestModelDuplicates.testDuplicatesForKeyPWN
import org.oewntk.model.LibTestModelDuplicates.testDuplicatesForKeyPos
import org.oewntk.xml.`in`.LibTestsXmlCommon.model
import org.oewntk.xml.`in`.LibTestsXmlCommon.ps

class TestsXmlModelDuplicates {

	@Test
	fun testKeyOEWN() {
		testDuplicatesForKeyOEWN(model!!, ps)
	}

	@Test
	fun testKeyPos() {
		testDuplicatesForKeyPos(model!!, ps)
	}

	@Test
	fun testKeyIC() {
		testDuplicatesForKeyIC(model!!, ps)
	}

	@Test
	fun testKeyPWN() {
		testDuplicatesForKeyPWN(model!!, ps)
	}

	companion object {

		@JvmStatic
		@BeforeClass
		fun init() {
			LibTestsXmlCommon.init()
			checkNotNull(model)
		}
	}
}
